apps/dcos_dns/src/dcos_dns_handler.erl

```diff
@@ -111,39 +121,55 @@ internal_resolve(Protocol, DNSMessage) ->
     Response = erldns_handler:do_handle(DNSMessage, ?LOCALHOST),
     encode_message(Protocol, Response).
 
--spec(resolve(protocol(), [upstream()], binary(), reply_fun()) -> ok).
-resolve(Protocol, Upstreams, Request, Fun) ->
+-spec(resolve(
```

Comment:
Not sure this is the best formatting, all the arguments alone are more than 80 columns.